### PR TITLE
Redmine 12537/adding ep ii expansion

### PR DESCRIPTION
**Pull request by bethsampher**
This adds a Freeze Plate column to the Well Details table on EP Pipeline II plates which allows users to input the number of child wells required and press create. The correct EP Pick and freeze plates are then created.

### CODE REVIEW

File: lib/LIMS2/Model/FormValidator/Constraint.pm [hunk 1]
```diff
@@ -470,6 +470,23 @@ sub existing_preset_id {
...
+    my $correct_name = $self->regexp_matches(qr/^HUPEP\d+$/); 
+    return sub {
+        my $value = shift;
+        return $exists->($value) && $self->model->schema->resultset('Plate')->find( { name => $value } )->type_id eq 'EP_PIPELINE_II' && $correct_name->($value);
+    };
+}
+
+
+sub ep_well {
+    return shift->regexp_matches(qr/^A0?\d+$/);
```
Comment (PeterFCKeen) [inline lib/LIMS2/Model/FormValidator/Constraint.pm]: Doesn't ep wells go A1-12 then B1-12?

File: lib/LIMS2/Model/Util/EPPipelineIIWellExpansion.pm [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+        $letter_id++;
+    }
+    return \@freeze_plates_created;
+}
+
+sub create_96_well_plates {
+    my $model             = shift;
+    my $letter_id         = shift;
+    my $plate_well_number = shift;
+    my $validated_params  = shift;
```
Comment (PeterFCKeen) [inline lib/LIMS2/Model/Util/EPPipelineIIWellExpansion.pm]: Styling - place the multiple parameters into an anonymous array format.
my ( $model, $letter_id, ..  ) = @_;

File: lib/LIMS2/Model/FormValidator/Constraint.pm (hunk 1 above)
Comment (bethsampher) [reply lib/LIMS2/Model/FormValidator/Constraint.pm]: Just double checked this and it is A01-A16

File: lib/LIMS2/Model/Util/EPPipelineIIWellExpansion.pm (hunk 2 above)
Comment (bethsampher) [reply lib/LIMS2/Model/Util/EPPipelineIIWellExpansion.pm]: Yes, I think it would make it clearer to change this

**Pull request merged**